implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -103,7 +105,18 @@ public String getValue(final String propertyName) {
     }
 
     boolean hasPropertyName(final String propertyName) {
-        return envVars.getEnv().containsKey(new EnvName(propertyName));
+        EnvName envName = reusableEnvName;
+        if (envName == null) {
+            envName = new EnvName(propertyName);
+            this.reusableEnvName = envName;
+        } else {
+            envName.setName(propertyName);
+        }
+        try {
+            return envVars.getEnv().containsKey(envName);
+        } finally {
+            envName.reset();
+        }
```

Comment:
This can cause issues if not thread-safe. While we use a simple `HashMap`, we never modify it after creation.